lib/std/map.zig

```diff
@@ -0,0 +1,872 @@
+            try testing.expect(!map.contains(key));
+        } else {
+            try testing.expect(map.contains(key));
+            try testing.expectEqual(key, map.get(key).?);
+        }
+    }
+}
+
+test "Map iterator after modifications" {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+    const allocator = gpa.allocator();
+
+    var map = Map(i32, i32).init(allocator);
+    defer map.deinit();
+
+    try map.insert(5, 50);
+    try map.insert(3, 30);
+    try map.insert(7, 70);
+    try map.insert(1, 10);
+    try map.insert(9, 90);
+
+    try testing.expect(map.erase(3));
+
+    var iter = map.iterator();
+    var count: usize = 0;
+    var sum: i32 = 0;
+
+    while (iter.next()) |entry| {
+        count += 1;
+        sum += entry.key;
+    }
+
+    try testing.expectEqual(@as(usize, 4), count);
+    try testing.expectEqual(@as(i32, 1 + 5 + 7 + 9), sum);
+}
+
+test "Map edge cases - single element operations" {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+    const allocator = gpa.allocator();
+
+    var map = Map(i32, i32).init(allocator);
+    defer map.deinit();
+
+    try map.insert(42, 84);
+    try testing.expectEqual(@as(usize, 1), map.size());
+    try testing.expectEqual(@as(i32, 84), map.get(42).?);
+
+    var iter = map.iterator();
+    const entry = iter.next().?;
+    try testing.expectEqual(@as(i32, 42), entry.key);
+    try testing.expectEqual(@as(i32, 84), entry.value);
+    try testing.expect(iter.next() == null);
+
+    try testing.expect(map.erase(42));
+    try testing.expectEqual(@as(usize, 0), map.size());
+    try testing.expect(map.empty());
+}
+
+test "Map duplicate key handling" {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+    const allocator = gpa.allocator();
+
+    var map = Map([]const u8, i32).init(allocator);
+    defer map.deinit();
+
+    try map.insert("duplicate", 1);
+    try map.insert("duplicate", 2);
+    try map.insert("duplicate", 3);
+
+    try testing.expectEqual(@as(usize, 1), map.size());
+    try testing.expectEqual(@as(i32, 3), map.get("duplicate").?);
+}
+
+test "Map with different value types" {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+    const allocator = gpa.allocator();
+
+    var bool_map = Map([]const u8, bool).init(allocator);
+    defer bool_map.deinit();
+
+    try bool_map.insert("true_key", true);
+    try bool_map.insert("false_key", false);
+
+    try testing.expect(bool_map.get("true_key").?);
+    try testing.expect(!bool_map.get("false_key").?);
+
+    var float_map = Map([]const u8, f64).init(allocator);
+    defer float_map.deinit();
+
+    try float_map.insert("pi", 3.14159);
+    try float_map.insert("e", 2.71828);
+
+    try testing.expectApproxEqRel(@as(f64, 3.14159), float_map.get("pi").?, 0.00001);
+    try testing.expectApproxEqRel(@as(f64, 2.71828), float_map.get("e").?, 0.00001);
```

Comment:
Are these approximate equals necessary? I think they should be used when doing computation on floats, but in this case i would expect to get the byte-for-byte same output as my input.